File: mock_data.py
"""Sample project data used in demo mode and in tests, matching the
normalized schema produced by data_transform.py."""

import pandas as pd


def sample_tasks() -> pd.DataFrame:
    return pd.DataFrame(
        [
            {"title": "Define project charter", "status": "Done", "assigned_to": "Alex Rivera",
             "start_date": "2026-07-01", "due_date": "2026-07-10", "percent_complete": 1.0, "priority": "High"},
            {"title": "Stakeholder kickoff", "status": "Done", "assigned_to": "Alex Rivera",
             "start_date": "2026-07-10", "due_date": "2026-07-14", "percent_complete": 1.0, "priority": "High"},
            {"title": "Requirements gathering", "status": "In Progress", "assigned_to": "Priya Shah",
             "start_date": "2026-07-14", "due_date": "2026-08-01", "percent_complete": 0.8, "priority": "High"},
            {"title": "Site migration plan", "status": "In Progress", "assigned_to": "Priya Shah",
             "start_date": "2026-07-20", "due_date": "2026-08-15", "percent_complete": 0.5, "priority": "Medium"},
            {"title": "Content inventory", "status": "In Progress", "assigned_to": "Jordan Lee",
             "start_date": "2026-07-25", "due_date": "2026-08-10", "percent_complete": 0.6, "priority": "Medium"},
            {"title": "Permissions mapping", "status": "Not Started", "assigned_to": "Jordan Lee",
             "start_date": "2026-08-05", "due_date": "2026-08-20", "percent_complete": 0.0, "priority": "Medium"},
            {"title": "Pilot migration batch", "status": "Not Started", "assigned_to": "Sam Okafor",
             "start_date": "2026-08-15", "due_date": "2026-08-25", "percent_complete": 0.0, "priority": "High"},
            {"title": "User training sessions", "status": "Not Started", "assigned_to": "Sam Okafor",
             "start_date": "2026-08-20", "due_date": "2026-09-05", "percent_complete": 0.0, "priority": "Low"},
            {"title": "Cutover checklist", "status": "Not Started", "assigned_to": "Alex Rivera",
             "start_date": "2026-08-01", "due_date": "2026-08-05", "percent_complete": 0.0, "priority": "High"},
            {"title": "Post-migration validation", "status": "Not Started", "assigned_to": "Priya Shah",
             "start_date": "2026-09-01", "due_date": "2026-09-10", "percent_complete": 0.0, "priority": "Medium"},
        ]
    ).assign(
        start_date=lambda df: pd.to_datetime(df["start_date"]),
        due_date=lambda df: pd.to_datetime(df["due_date"]),
    )


def sample_risks() -> pd.DataFrame:
    return pd.DataFrame(
        [
            {"title": "Legacy list templates not supported", "severity": "High", "owner": "Priya Shah",
             "status": "Open", "description": "Custom InfoPath forms have no direct SharePoint Online equivalent."},
            {"title": "Tenant storage quota", "severity": "Medium", "owner": "Sam Okafor",
             "status": "Open", "description": "Combined library size may exceed default site quota."},
            {"title": "Third-party workflow add-in", "severity": "High", "owner": "Jordan Lee",
             "status": "Open", "description": "Nintex workflows need to be rebuilt in Power Automate."},
            {"title": "User adoption resistance", "severity": "Low", "owner": "Alex Rivera",
             "status": "Mitigated", "description": "Early training sessions well received in pilot group."},
            {"title": "Downtime during cutover", "severity": "Medium", "owner": "Alex Rivera",
             "status": "Open", "description": "Cutover window needs off-hours scheduling."},
        ]
    )

File: aspx/App_Code/MockData.cs
using System;
using System.Collections.Generic;

namespace SharePointPmDashboard.App_Code
{
    /// <summary>Sample project data used in demo mode, matching mock_data.py in the Streamlit version.</summary>
    public static class MockData
    {
        public static List<TaskItem> GetSampleTasks()
        {
            return new List<TaskItem>
            {
                new TaskItem { Title = "Define project charter", Status = "Done", AssignedTo = "Alex Rivera",
                    StartDate = new DateTime(2026, 7, 1), DueDate = new DateTime(2026, 7, 10), PercentComplete = 1.0, Priority = "High" },
                new TaskItem { Title = "Stakeholder kickoff", Status = "Done", AssignedTo = "Alex Rivera",
                    StartDate = new DateTime(2026, 7, 10), DueDate = new DateTime(2026, 7, 14), PercentComplete = 1.0, Priority = "High" },
                new TaskItem { Title = "Requirements gathering", Status = "In Progress", AssignedTo = "Priya Shah",
                    StartDate = new DateTime(2026, 7, 14), DueDate = new DateTime(2026, 8, 1), PercentComplete = 0.8, Priority = "High" },
                new TaskItem { Title = "Site migration plan", Status = "In Progress", AssignedTo = "Priya Shah",
                    StartDate = new DateTime(2026, 7, 20), DueDate = new DateTime(2026, 8, 15), PercentComplete = 0.5, Priority = "Medium" },
                new TaskItem { Title = "Content inventory", Status = "In Progress", AssignedTo = "Jordan Lee",
                    StartDate = new DateTime(2026, 7, 25), DueDate = new DateTime(2026, 8, 10), PercentComplete = 0.6, Priority = "Medium" },
                new TaskItem { Title = "Permissions mapping", Status = "Not Started", AssignedTo = "Jordan Lee",
                    StartDate = new DateTime(2026, 8, 5), DueDate = new DateTime(2026, 8, 20), PercentComplete = 0.0, Priority = "Medium" },
                new TaskItem { Title = "Pilot migration batch", Status = "Not Started", AssignedTo = "Sam Okafor",
                    StartDate = new DateTime(2026, 8, 15), DueDate = new DateTime(2026, 8, 25), PercentComplete = 0.0, Priority = "High" },
                new TaskItem { Title = "User training sessions", Status = "Not Started", AssignedTo = "Sam Okafor",
                    StartDate = new DateTime(2026, 8, 20), DueDate = new DateTime(2026, 9, 5), PercentComplete = 0.0, Priority = "Low" },
                new TaskItem { Title = "Cutover checklist", Status = "Not Started", AssignedTo = "Alex Rivera",
                    StartDate = new DateTime(2026, 8, 1), DueDate = new DateTime(2026, 8, 5), PercentComplete = 0.0, Priority = "High" },
                new TaskItem { Title = "Post-migration validation", Status = "Not Started", AssignedTo = "Priya Shah",
                    StartDate = new DateTime(2026, 9, 1), DueDate = new DateTime(2026, 9, 10), PercentComplete = 0.0, Priority = "Medium" },
            };
        }

        public static List<RiskItem> GetSampleRisks()
        {
            return new List<RiskItem>
            {
                new RiskItem { Title = "Legacy list templates not supported", Severity = "High", Owner = "Priya Shah",
                    Status = "Open", Description = "Custom InfoPath forms have no direct SharePoint Online equivalent." },
                new RiskItem { Title = "Tenant storage quota", Severity = "Medium", Owner = "Sam Okafor",
                    Status = "Open", Description = "Combined library size may exceed default site quota." },
                new RiskItem { Title = "Third-party workflow add-in", Severity = "High", Owner = "Jordan Lee",
                    Status = "Open", Description = "Nintex workflows need to be rebuilt in Power Automate." },
                new RiskItem { Title = "User adoption resistance", Severity = "Low", Owner = "Alex Rivera",
                    Status = "Mitigated", Description = "Early training sessions well received in pilot group." },
                new RiskItem { Title = "Downtime during cutover", Severity = "Medium", Owner = "Alex Rivera",
                    Status = "Open", Description = "Cutover window needs off-hours scheduling." },
            };
        }
    }
}
